lib/cosmos/packets/packet_config.rb

```diff
@@ -657,25 +764,80 @@ def xtce_process_element(element, depth)
-          object = argument
-          type = argument_type
-        else
+      when 'ParameterRefEntry', 'ArgumentRefEntry', 'ArrayParameterRefEntry', 'ArrayArgumentRefEntry'
+        reference_location, bit_offset = xtce_handle_location_in_container_in_bits(element)
+
+        array_type = nil
+        array_bit_size = nil
+        if element.name =~ /Parameter/
           # Look up the parameter and parameter type
           parameter = @parameters[element['parameterRef']]
           raise "parameterRef #{element['parameterRef']} not found" unless parameter
           parameter_type = @parameter_types[parameter.parameterTypeRef]
           raise "parameterTypeRef #{parameter.parameterTypeRef} not found" unless parameter_type
+          if element.name == 'ArrayParameterRefEntry'
+            array_type = parameter_type
+            parameter_type = @parameter_types[array_type.arrayTypeRef]
+            raise "arrayTypeRef #{parameter.arrayTypeRef} not found" unless parameter_type
+          end
           refName = 'parameterRef'
           object = parameter
           type = parameter_type
+        else
+          # Look up the argument and argument type
+          if element.name == 'ArrayArgumentRefEntry'
+            # Requiring parameterRef for argument arrays appears to be a defect in the schema
+            argument = @arguments[element['parameterRef']]
+            raise "parameterRef #{element['parameterRef']} not found" unless argument
+            argument_type = @argument_types[argument.argumentTypeRef]
+            raise "argumentTypeRef #{argument.argumentTypeRef} not found" unless argument_type
+            array_type = argument_type
+            argument_type = @argument_types[array_type.arrayTypeRef]
+            raise "arrayTypeRef #{array_type.arrayTypeRef} not found" unless argument_type
+            refName = 'parameterRef'
+          else
+            argument = @arguments[element['argumentRef']]
+            raise "argumentRef #{element['argumentRef']} not found" unless argument
+            argument_type = @argument_types[argument.argumentTypeRef]
+            raise "argumentTypeRef #{argument.argumentTypeRef} not found" unless argument_type
+            refName = 'argumentRef'
+          end
+          object = argument
+          type = argument_type
+        end
+
+        bit_size = Integer(type.sizeInBits)
+
+        if array_type
+          array_num_items = 1
+          # Need to determine dimensions
+          xtce_recurse_element(element, depth + 1) do |element, depth|
+            if element.name == 'Dimension'
+              starting_index = 0
+              ending_index = 0
+              element.children.each do |child_element|
+                if child_element.name == 'StartingIndex'
+                  child_element.children.each do |child_element2|
+                    if child_element2.name == 'FixedValue'
+                      starting_index = child_element2.text.to_i
+                    end
```

Comment:
Basically ignoring anything else at this point.   For this example there shouldn't be anything else.